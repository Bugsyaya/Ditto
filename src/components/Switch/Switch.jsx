import React from 'react';
import { Switch as AntdSwitch } from 'antd'

const Switch = ({
  props,
  children,
  /**
  * Level of the Switch defining it's color and outline.
  * Possible values are: primary, secondary, tertiary, warning & danger.
  */
  autoFocus = false,
  checked = false,
  checkedChildren,
  className,
  defaultChecked = false,
  disabled = false,
  loading = false,
  size = 'default',
  unCheckedChildren,
  onChange,
  onClick
}) => {
  return <AntdSwitch
    aria-label='switch'
    autoFocus={autoFocus}
    checked={checked}
    checkedChildren={checkedChildren}
    className={className}
    defaultChecked={defaultChecked}
    disabled={disabled}
    loading={loading}
    size={size}
    unCheckedChildren={unCheckedChildren}
    onChange={onChange}
    onClick={onClick}
    {...props} />
}

export { Switch };