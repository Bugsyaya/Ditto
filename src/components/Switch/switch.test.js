import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import React from 'react';
import {Switch} from './Switch';

describe('Switch component', () => {
  const swi = (<Switch/>);

  it('Snapshot button', () => {
    const { baseElement } = render(swi);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render given children', () => {
    render(swi);
    expect(screen.getByLabelText('switch')).toBeInTheDocument();
  });
});