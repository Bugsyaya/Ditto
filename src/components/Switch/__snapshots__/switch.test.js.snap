// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component Snapshot button 1`] = `
<body>
  <div>
    <button
      aria-checked="false"
      aria-label="switch"
      class="ant-switch"
      role="switch"
      type="button"
    >
      <div
        class="ant-switch-handle"
      />
      <span
        class="ant-switch-inner"
      />
    </button>
  </div>
</body>
`;
